### PR TITLE
chore(launch): add more error and warning coverage

### PR DESCRIPTION
**Pull request by gtarpenning**
Fixes
-----
https://wandb.atlassian.net/browse/WB-15807

Description
-----------
TODO(gst): Make warnings/errors more actionable. This is tricky because most of these errors come from badly formatted metadata, which can technically be edited by the user. So it could just be a case of a user accidentally deleting or manipulating metadata...

Surfaces more error conditions, mostly as warnings, in the case where wandb runs don't intend on creating jobs. 

### <samp>🤖 Generated by Copilot at 263c0e3</samp>

This pull request improves the user experience of creating and launching jobs with wandb by adding more informative warnings and error messages. It also updates the `job_builder` and `create_job` modules to handle some edge cases and exceptions.

Testing
-------
How was this PR tested?

Checklist
-------
- [ ] Include reference to internal ticket "Fixes WB-NNNNN" and/or GitHub issue "Fixes #NNNN" (if applicable)
- [ ] Ensure PR title compliance with the [conventional commits standards](https://github.com/wandb/wandb/blob/main/CONTRIBUTING.md#conventional-commits)

### <samp>🤖 Generated by Copilot at 263c0e3</samp>

> _When creating a job artifact_
> _You might encounter some termerror_
> _To help you debug_
> _We added some warnings_
> _And pointed you to the docs for support_

### CODE REVIEW

**Comment by gtarpenning** [comment]
Screen Shot 2023-10-10 at 9 25 00 PM
[link](https://app.circleci.com/pipelines/github/wandb/wandb/27622/workflows/641940bc-a0d7-4d09-a342-4ed4886de4df/jobs/830997/tests)

This is an example of why adding warnings is not quite as trivial as it might seem. This test will either have to be changed (to accept a warning as correct? 👎🏻 ) this warning removed, or we create more complex process of inferring the intent of the run... 

I think its possible that all runs will now display **some** sort of warning, which i'd imagine would be quite alarming for most users.

File: wandb/sdk/internal/job_builder.py [hunk 1]
```diff
@@ -296,14 +299,23 @@ def build(self) -> Optional[Artifact]:
...
             os.path.join(self._settings.files_dir, REQUIREMENTS_FNAME)
         ):
+            wandb.termwarn(
+                "No requirements.txt found, not creating job artifact. See https://docs.wandb.ai/guides/launch/create-job"
+            )
             return None
         metadata = self._handle_metadata_file()
         if metadata is None:
+            wandb.termwarn(
+                "No metadata file found, not creating job artifact. Ensure api settings files_dir exists with read and write access."
```
Comment (KyleGoyette) [inline wandb/sdk/internal/job_builder.py:309]: I dont think a user could do much with this warning. "Ensure read and write access to run files dir: {self._settings.files_dir}, control this via the WANDB_DIR env var" or something?

**Review by KyleGoyette — APPROVED** [review]
One warning to improve IMO

File: wandb/sdk/internal/job_builder.py (hunk 1 above)
Comment (KyleGoyette) [reply wandb/sdk/internal/job_builder.py:309]: Maybe even link to the env var docs...

**Pull request merged**